controller/controller.go

```diff
@@ -53,175 +55,191 @@ func NewControllerWithClient(logger *log.Logger, client resources.StorageClient,
 	return &Controller{logger: logger, Client: client, exec: exec}
 }
 
-//Init method is to initialize the flexvolume
-func (c *Controller) Init(config resources.UbiquityPluginConfig) resources.FlexVolumeResponse {
+//Init method is to initialize the k8sresourcesvolume
+func (c *Controller) Init(config resources.UbiquityPluginConfig) k8sresources.FlexVolumeResponse {
 	c.logger.Println("controller-activate-start")
 	defer c.logger.Println("controller-activate-end")
 
 	activateRequest := resources.ActivateRequest{Backends: config.Backends}
 	err := c.Client.Activate(activateRequest)
 	if err != nil {
-		return resources.FlexVolumeResponse{
+		return k8sresources.FlexVolumeResponse{
 			Status:  "Failure",
 			Message: fmt.Sprintf("Plugin init failed %#v ", err),
-			Device:  "",
 		}
 
 	}
 
-	return resources.FlexVolumeResponse{
+	return k8sresources.FlexVolumeResponse{
 		Status:  "Success",
 		Message: "Plugin init successfully",
-		Device:  "",
 	}
 }
 
 //Attach method attaches a volume to a host
-func (c *Controller) Attach(attachRequest map[string]string) resources.FlexVolumeResponse {
+func (c *Controller) Attach(attachRequest k8sresources.FlexVolumeAttachRequest) k8sresources.FlexVolumeResponse {
 	c.logger.Println("controller-attach-start")
 	defer c.logger.Println("controller-attach-end")
-	c.logger.Printf("attach-details %#v\n", attachRequest)
-	var attachResponse resources.FlexVolumeResponse
-	volumeName, exists := attachRequest["volumeName"]
-	if !exists {
 
-		attachResponse = resources.FlexVolumeResponse{
-			Status:  "Failure",
-			Message: fmt.Sprintf("Failed to attach volume: VolumeName not found : #%v", attachRequest),
-			Device:  volumeName,
+	if attachRequest.Version == k8sresources.KubernetesVersion_1_5 {
+		c.logger.Printf("k8s version 1.5")
+		return k8sresources.FlexVolumeResponse{
+			Status: "Success",
 		}
-		c.logger.Printf("Failed-to-attach-volume, VolumeName found %#v ", attachRequest)
-		return attachResponse
-
 	}
+	c.logger.Printf("k8s version 1.6 or later")
+	return k8sresources.FlexVolumeResponse{
+		Status: "Not supported",
+	}
+}
 
-	getVolumeRequest := resources.GetVolumeRequest{Name: volumeName}
-	_, err := c.Client.GetVolume(getVolumeRequest)
-
-	if err != nil {
-		return resources.FlexVolumeResponse{
-			Status:  "Failure",
-			Message: "Failed checking volume, call create before attach",
-			Device:  volumeName}
+//GetVolumeName checks if volume is attached
+func (c *Controller) GetVolumeName(getVolumeNameRequest k8sresources.FlexVolumeGetVolumeNameRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-isAttached-start")
+	defer c.logger.Println("controller-isAttached-end")
 
+	return k8sresources.FlexVolumeResponse{
+		Status: "Not supported",
```

Comment:
We avoid a known bug of pv spec reconstruction:
https://github.com/kubernetes/community/pull/650